services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDocumentBuilder.java

```diff
@@ -46,6 +48,22 @@ public MongoDbDocumentBuilder withDeviceId(final String deviceId) {
         return this;
     }
 
+    /**
+     * Adds nested CA subjectDn property.
+     *
+     * @param entityPropKey key where entity properties are stored. e.g.: "tenant" for
+     *                      the {@link org.eclipse.hono.service.management.tenant.Tenant}
+     * @param caSubjectDn certificate Authority subjectDn
+     * @return a reference to this for fluent use.
+     */
+    public MongoDbDocumentBuilder withCa(final String entityPropKey, final String caSubjectDn) {
```

Comment:
The`entityPropKey` is always a constant and probably you could remove this parameter. Within the method `withCa` `RegistryManagementConstants.FIELD_TENANT`  could be used instead of string literal "tenant".

Reply:
done